File: cpp/Day11/demo14/src/employee.cpp
#include "../include/employee.h"
#include "../include/invalid_data_exception.h"
Employee::Employee()
{
    this->empid = 0;
    this->salary = 0;
}
Employee::Employee(int empid, string name, double salary) : Person(name)
{
    this->empid = empid;
    this->salary = salary;
}
void Employee::acceptData()
{
    cout << "Enter empid = ";
    cin >> this->empid;
    Person::acceptData();
    cout << "Enter salary = ";
    cin >> this->salary;
}
void Employee::printData()
{
    cout << "Empid = " << this->empid << endl;
    cout << "Name = " << this->getName() << endl;
    cout << "Salary = " << this->salary << endl;
}
void Employee::setEmpid(int empid)
{
    this->empid = empid;
}
void Employee::setSalary(double salary)
{
    this->salary = salary;
}
int Employee::getEmpid()
{
    return this->empid;
}
double Employee::getSalary()
{
    return this->salary;
}

File: Core Java/oopjava/Day05/Demo03/src/com/sunbeam/Employee.java
package com.sunbeam;

import java.util.Scanner;

public class Employee extends Person // inheritance
{
	int empid;
	double salary;

	public Employee() {
		System.out.println("Employee Parameterless Ctor");
	}

	public Employee(int empid, double salary) {
		System.out.println("Employee Parameterized Ctor-1");
		this.empid = empid;
		this.salary = salary;
	}

	public Employee(int empid,String name, String mobile, double salary) {
		super(name, mobile);
		System.out.println("Employee Parameterized Ctor-2");
		this.empid = empid;
		this.salary = salary;
	}

	@Override
	public void acceptData() {
		Scanner sc = new Scanner(System.in);
		System.out.print("Enter Empid - ");
		empid = sc.nextInt();
		
		super.acceptData();
		
		System.out.print("Enter Salary - ");
		salary = sc.nextDouble();
	}
	
	@Override
	public void displayData() {
		System.out.println("Empid - " + empid);
		super.displayData();
		System.out.println("Salary - " + salary);
	}
	
	

}
